### ISSUE TITLE
No output unless verbose flag is enabled

### ISSUE THREAD

**Issue by keeganstreet** [sasstools/sass-lint#864]
No output is generated unless I use the verbose flag.

Version: 1.9.1

**Results with verbose flag (note one error and one warning)**

```
> sass-lint --verbose


source/assets/scss/style.scss
   7:2  warning  Vendor prefixes should not be used  no-vendor-prefixes
  16:1  error    ID selectors not allowed            no-ids

✖ 2 problems (1 error, 1 warning)

/Users/kestreet/Projects/Deloitte/Middleman/node_modules/sass-lint/index.js:294
    throw new Error(errorCount.count + ' errors were detected in \n- ' + errorCount.files.join('\n- '));
    ^

Error: 1 errors were detected in 
- source/assets/scss/style.scss
    at Function.sassLint.failOnError (/Users/kestreet/Projects/Deloitte/Middleman/node_modules/sass-lint/index.js:294:11)
    at detectPattern (/Users/kestreet/Projects/Deloitte/Middleman/node_modules/sass-lint/bin/sass-lint.js:33:10)
    at Object.<anonymous> (/Users/kestreet/Projects/Deloitte/Middleman/node_modules/sass-lint/bin/sass-lint.js:94:3)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Function.Module.runMain (module.js:441:10)
    at startup (node.js:134:18)
    at node.js:962:3
```

**Results without verbose flag**

```
> sass-lint

/Users/kestreet/Projects/Deloitte/Middleman/node_modules/sass-lint/index.js:294
    throw new Error(errorCount.count + ' errors were detected in \n- ' + errorCount.files.join('\n- '));
    ^

Error: 1 errors were detected in 
- source/assets/scss/style.scss
    at Function.sassLint.failOnError (/Users/kestreet/Projects/Deloitte/Middleman/node_modules/sass-lint/index.js:294:11)
    at detectPattern (/Users/kestreet/Projects/Deloitte/Middleman/node_modules/sass-lint/bin/sass-lint.js:33:10)
    at Object.<anonymous> (/Users/kestreet/Projects/Deloitte/Middleman/node_modules/sass-lint/bin/sass-lint.js:94:3)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Function.Module.runMain (module.js:441:10)
    at startup (node.js:134:18)
    at node.js:962:3
```

**Results without verbose flag, with no-exit flag**

```
> sass-lint --no-exit


```

The readme states that:

> Warnings or any rule set to severity: `1` in your config by default will not be reported by the CLI tool unless you use verbose flag `-v` or `--verbose`.

As you can see in result 1, there is an error and a warning in the processed CSS. I would expect that the warning is suppressed when I don't use the verbose flag, but the error should still be reported.

**Comment by DanPurdy** [comment]
Sorry for taking so long to get back to this one. I think the whole verbose flag is just confusing. The behaviour you are seeing is as I would expect though. 

The output is being supressed but the thrown error is not being caught in your 2nd example. If you combine the `--no-exit` without `--verbose` then you would have the silent output as you have in the final example.

Verbose in this sense (and it was implemented long before I came along) is just the printed output rather than reporting why a exit code 1 was thrown.

Would be happy to discuss ways to make this clearer and It's definitely an intention of mine to clear this functionality up / move it inline with other Linter CLI's for our V2 release.
